### ISSUE TITLE
`Layout/MultilineArrayLineBreaks` tests missing

### ISSUE THREAD

**Issue by mlwyatt** [rubocop/rubocop#11797]
### Discussed in https://github.com/rubocop/rubocop/discussions/11771

<div type='discussions-op-text'>

<sup>Originally posted by **mlwyatt** April  6, 2023</sup>
The tests in [`multiline_array_line_breaks_spec.rb` ](https://github.com/rubocop/rubocop/blob/f3255797ee094c69fe4ae4bf5315d601ed182cdb/spec/rubocop/cop/layout/multiline_array_line_breaks_spec.rb#L54) work perfectly when `AllowMultilineFinalElement` is `true`.
However, there's no test for when `AllowMultilineFinalElement` is `false` to trigger an offense for the code below, even though the docs list it as a "bad" example.

To be clear, when running `rubocop` against this code, it also works perfectly. 

```ruby
[a, b, foo(
  bar
)]
```
</div>

**Comment by koic** [comment]
I'm not sure if it's better, but the default is `AllowMultilineFinalElement: false`, test cases that aren't `AllowMultilineFinalElement: true` can be considered `AllowMultilineFinalElement: false`.

OTOH, I've updated the docs where it was confusing about the config parameter:
https://github.com/rubocop/rubocop/commit/8dfe1b473d8e6083dded2c10cba0da6ab2354e69
